hcloud/resource_hcloud_reversedns.go

```diff
@@ -0,0 +1,262 @@
+		id, ok = d.GetOk("floating_ip_id")
+		if !ok {
+			log.Printf("[WARN] Invalid id (%s), removing from state: %v", d.Id(), ok)
+			d.SetId("")
+			return nil
+		}
+		floatingIP, _, _ := client.FloatingIP.GetByID(ctx, id.(int))
+		d.SetId(generateRDNSID(nil, floatingIP, ip))
+		rdnsType = "floatingIP"
+	} else {
+		server, _, _ := client.Server.GetByID(ctx, id.(int))
+		d.SetId(generateRDNSID(server, nil, ip))
+		rdnsType = "server"
+	}
+
+	switch rdnsType {
+	case "floating_ip":
+		floatingIP, _, err := client.FloatingIP.GetByID(ctx, id.(int))
+		if err != nil {
+			return err
+		}
+		if floatingIP == nil {
+			log.Printf("[WARN] Floating IP (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		action, _, err := client.FloatingIP.ChangeDNSPtr(ctx, floatingIP, ip, &ptr)
+		if err != nil {
+			return err
+		}
+		if err := waitForFloatingIPAction(ctx, client, action, floatingIP); err != nil {
+			return err
+		}
+	case "server":
+		server, _, err := client.Server.GetByID(ctx, id.(int))
+		if err != nil {
+			return err
+		}
+		if server == nil {
+			log.Printf("[WARN] Server (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		action, _, err := client.Server.ChangeDNSPtr(ctx, server, ip, &ptr)
+		if err != nil {
+			return err
+		}
+		if err := waitForServerAction(ctx, client, action, server); err != nil {
+			return err
+		}
+
+	}
+	return resourceReverseDNSRead(d, m)
+}
+func resourceReverseDNSUpdate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*hcloud.Client)
+	ctx := context.Background()
+	server, floatingIP, err := lookupRDNSID(d.Id(), client)
+	if err != nil {
+		return err
+	}
+	ip := d.Get("ip_address").(string)
+	ptr := d.Get("dns_ptr").(string)
+
+	if floatingIP != nil {
+		if d.HasChange("dns_ptr") {
+			if err != nil {
+				return err
+			}
+			if floatingIP == nil {
+				log.Printf("[WARN] Floating IP (%s) not found, removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+			action, _, err := client.FloatingIP.ChangeDNSPtr(ctx, floatingIP, ip, &ptr)
+			if err != nil {
+				return err
+			}
+			if err := waitForFloatingIPAction(ctx, client, action, floatingIP); err != nil {
+				return err
+			}
+		}
+	} else if server != nil {
+		if err != nil {
+			return err
+		}
+		if server == nil {
+			log.Printf("[WARN] Server (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		action, _, err := client.Server.ChangeDNSPtr(ctx, server, ip, &ptr)
+		if err != nil {
+			return err
+		}
+		if err := waitForServerAction(ctx, client, action, server); err != nil {
+			return err
+		}
+
+	}
+	return resourceReverseDNSRead(d, m)
+}
+func resourceReverseDNSDelete(d *schema.ResourceData, m interface{}) error {
+	client := m.(*hcloud.Client)
+	ctx := context.Background()
+	server, floatingIP, err := lookupRDNSID(d.Id(), client)
+	if err != nil {
+		log.Printf("[WARN] Invalid id (%s), removing from state: %s", d.Id(), err)
```

Comment:
If `lookupRDNSID` returns an error you have cover two cases:
1. Invalid id => log invalid id, remove from state and return nil
2. every other error => directly return the error

Please also make sure the errors of `lookupRNDSID` are handled the same way everywhere the function is used. (line 161)